presto-main/src/main/java/io/prestosql/sql/planner/RelationPlanner.java

```diff
@@ -192,6 +194,40 @@ private RelationPlan addRowFilters(Table node, RelationPlan plan)
         return new RelationPlan(planBuilder.getRoot(), plan.getScope(), plan.getFieldMappings());
     }
 
+    private RelationPlan addColumnMasks(Table table, RelationPlan plan)
+    {
+        Map<String, List<Expression>> columnMasks = analysis.getColumnMasks(table);
+
+        PlanNode root = plan.getRoot();
+        List<Symbol> mappings = plan.getFieldMappings();
+
+        TranslationMap translations = new TranslationMap(plan, analysis, lambdaDeclarationToSymbolMap);
+        translations.setFieldMappings(mappings);
+
+        PlanBuilder planBuilder = new PlanBuilder(translations, root);
+
+        for (int i = 0; i < plan.getDescriptor().getAllFieldCount(); i++) {
+            Field field = plan.getDescriptor().getFieldByIndex(i);
+
+            for (Expression mask : columnMasks.getOrDefault(field.getName().get(), ImmutableList.of())) {
```

Comment:
Add a test where there are two column masks (one connector level and one system level) and where we could notice the order in which they were applied.

Reply:
Unfortunately, we don't have infrastructure to do that. It would require implementing masking support in real connectors. The tests rely on simulating the ordering of the masks by explicitly registering them in TestingAccessControl

Reply:
Then I think we need to write `TestAccessControl` unit tests where you could inject some fake system and connector access control to test (concrete) the order.

Reply:
Do you have a suggestion about where to do it! I don’t believe we have anything where we can inject fake connectors like that and still be able to run reasonable queries.

Reply:
Maybe we could add column mask support for file based access controls, then we could use both of them in single test.

Reply:
That’s an option, but it increases the scope of this change (we need to design how we want to represent them in the files, etc)

Reply:
Another option is to add a test to `io.prestosql.security.TestAccessControlManager` where you could inject your custom plugins, like: https://github.com/prestosql/presto/pull/2924/files#diff-969c42278114183bfc24a2cde29f7289R200

Reply:
That kind of test is super brittle. But ok, added it.